src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -149,6 +149,17 @@ static IEnumerable<FunctionOverload> GetAlwaysPermittedOverloads()
                     .WithRequiredParameter("network", LanguageConstants.String, "String in CIDR notation containing an IP address range to be converted.")
                     .Build();
 
+                yield return new FunctionOverloadBuilder("buildUri")
+                    .WithReturnResultBuilder(TryDeriveLiteralReturnType("buildUri", LanguageConstants.String), LanguageConstants.String).WithGenericDescription("Constructs a URI from specified components and returns an object with these components.")
+                    .WithRequiredParameter("components", GetParseOrBuildUriReturnType(), "An object containing URI components such as scheme, host, port, path, and query.")
+                    .Build();
+
+                yield return new FunctionOverloadBuilder("parseUri")
+                    .WithReturnType(GetParseOrBuildUriReturnType())
```

Comment:
You can use `TryDeriveLiteralReturnType` for `parseUri`, too.

Reply:
Could you add one or two tests to https://github.com/Azure/bicep/blob/688cd8f42cb2100031b60a9de7a4492e0fb5bff9/src/Bicep.Core.UnitTests/TypeSystem/FunctionResolverTests.cs#L843 for both `buildUri` and `parseUri`? These can be copied from the tests in the backend codebase.